src/jit/importer.cpp

```diff
@@ -1490,16 +1490,21 @@ var_types Compiler::impNormStructType(CORINFO_CLASS_HANDLE structHnd,
     var_types   structType  = TYP_STRUCT;
 
 #ifdef FEATURE_CORECLR
-    const bool hasGCPtrs = (structFlags & CORINFO_FLG_CONTAINS_GC_PTR) != 0;
+    // On coreclr the check for GC includes a "may" to account for the special 
+    // ByRef like span structs.  The added check for "CONTAINS_STACK_PTR" is the particular bit.
+    // When this is set the struct will contain a ByRef that could be a GC pointer or a native
+    // pointer.
+    const bool mayContainGCPtrs = ((structFlags & CORINFO_FLG_CONTAINS_STACK_PTR) != 0 
+                                    || ((structFlags & CORINFO_FLG_CONTAINS_GC_PTR) != 0));
 #else
     // Desktop CLR won't report FLG_CONTAINS_GC_PTR for RefAnyClass - need to check explicitly.
```

Comment:
You should not need this ifdef for desktop anymore. Desktop will return `CORINFO_FLG_CONTAINS_STACK_PTR` for TypeRef and so the code under the `#ifdef FEATURE_CORECLR` will work just fine even on desktop.